src/node_modules.cc

```diff
@@ -0,0 +1,280 @@
+#include "node_modules.h"
+#include "base_object-inl.h"
+#include "node_errors.h"
+#include "node_external_reference.h"
+#include "util-inl.h"
+#include "v8-fast-api-calls.h"
+#include "v8.h"
+
+namespace node {
+namespace modules {
+
+using errors::TryCatchScope;
+
+using v8::CFunction;
+using v8::Context;
+using v8::FastOneByteString;
+using v8::FunctionCallbackInfo;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::JSON;
+using v8::Local;
+using v8::MaybeLocal;
+using v8::NewStringType;
+using v8::Object;
+using v8::ObjectTemplate;
+using v8::String;
+using v8::Value;
+
+void BindingData::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("package_json_files", package_json_files_);
+}
+
+BindingData::BindingData(Realm* realm, v8::Local<v8::Object> object)
+    : SnapshotableObject(realm, object, type_int) {
+  package_json_files_ =
+      v8::Global<v8::Map>(realm->isolate(), v8::Map::New(realm->isolate()));
```

Comment:
Instead of creating it again during deserialization, I think you can just call `package_json_files_ .Clear()` in `PrepareForSerialization()` and record a reference to this map using the `creator->AddData(context, ..)`, then at deserialization time get it back via `context->GetDataFromSnapshotOnce(index)`. See the v8 or encoding binding data for an example.